service/controller/chart/v1/resource/tiller/delete.go

```diff
@@ -0,0 +1,10 @@
+package tiller
+
+import (
+	"context"
+)
+
+// EnsureDeleted is not implemented for the tiller resource.
+func (r *Resource) EnsureDeleted(ctx context.Context, obj interface{}) error {
+	return nil
```

Comment:
I think we should ensure the installation here as well. Once use case would be an upgrade necessary to perform the action at all. Another would be a tenant cluster creation that is not completed but the tenant cluster got requested to be deleted. Then some code could fail down the road during deletion.

Reply:
Good call. I implemented both methods and moved them to resource.go.